hilti/toolchain/src/compiler/visitors/resolver.cc

```diff
@@ -53,23 +53,6 @@ struct Visitor : public visitor::PostOrder<void, Visitor> {
     bool modified = false;
     std::map<ID, Type> auto_params; // miapping of `auto` parameters inferred, indexed by canonical ID
 
-#if 0
```

Comment:
These are left in on purpose, to activate for debugging. 

More, generally don't like disallowing `#if 0`, suggest to drop `0aa5ba4908d` and suppress the warning instead.

Reply:
I checked the code I removed again and if activated none of the removed code would compile today. This is exactly one reason [this lint](https://clang.llvm.org/extra/clang-tidy/checks/readability/avoid-unconditional-preprocessor-if.html) flagged this an an antipattern:

> Unconditional preprocessor directives, such as `#if 0` for disabled code and `#if 1` for enabled code, can lead to dead code and always enabled code, respectively. Dead code can make understanding the codebase more difficult, hinder readability, and may be a sign of unfinished functionality or abandoned features. This can cause maintenance issues, confusion for future developers, and **potential compilation problems**.

A workable approach they suggest would be to use some debugging define which would allow actually enabling this usually disabled code in e.g., CI to make sure it keep working:

> As a solution for both cases, consider using preprocessor macros or defines, like `#ifdef DEBUGGING_ENABLED`, to control code enabling or disabling. This approach provides better coordination and flexibility when working with different parts of the codebase. Alternatively, you can comment out the entire code using `/* */` block comments and add a hint, such as @todo, to indicate future actions.

We could consider following that suggestion, and add a setup in CI which enables this code and fix the issues the code currently has. I personally never enabled any of these snippets so I am unsure how much value that would provide though.

Reply:
I believe this is overthinking it ... The `#if 0` is just there to save the code in case it's helpful again later, rather than having to come up with it once more then. If it doesn't compile anymore, fine; still better to have a starting point than nothing (I have used some of it multiple times). However, in the grand scheme of things the code is really not important, so custom `#define`, CI, etc. seems too much. 

But if it's better to use comments instead of `#define`, sure, I wont' fight that. :-)